mdal/frmts/mdal_flo2d.cpp

```diff
@@ -119,14 +119,342 @@ void MDAL::DriverFlo2D::parseCADPTSFile( const std::string &datFileName, std::ve
+    {
+      for ( size_t i = 0; i < data.size(); ++i )
+      {
+        if ( data[i].size() != datasetGroups.size() )
+          throw MDAL::Error( MDAL_Status::Err_UnknownFormat, "Error while reading HYCHAN file, wrong format" );
+
+        for ( size_t j = 0; j < data[i].size(); ++j )
+        {
+          if ( i >= datasetGroups[j]->datasets.size() )
+            throw MDAL::Error( MDAL_Status::Err_UnknownFormat, "Error while reading HYCHAN file, wrong format" );
+
+          static_cast<MDAL::MemoryDataset2D *>( datasetGroups[j]->datasets[i].get() )->setScalarValue( it->second, data[i][j] );
+        }
+      }
+    }
+  }
+
+  for ( std::shared_ptr<DatasetGroup> datasetGroup : datasetGroups )
+  {
+    for ( std::shared_ptr<Dataset> dataset : datasetGroup->datasets )
+      dataset->setStatistics( MDAL::calculateStatistics( dataset ) );
+
+    datasetGroup->setStatistics( MDAL::calculateStatistics( datasetGroup ) );
+    mMesh->datasetGroups.push_back( datasetGroup );
+  }
+}
+
+
+void MDAL::DriverFlo2D::createMesh1d( const std::string &datFileName, const std::vector<MDAL::DriverFlo2D::CellCenter> &cells, std::map<size_t, size_t> &cellsIdToVertex )
+{
+  std::map<size_t, std::vector<size_t>> duplicatedRightBankToVertex;
+  std::vector<Vertex> vertices;
+  std::vector<Edge> edges;
+  size_t verticesCount;
+
+  parseCHANBANKFile( datFileName, cellsIdToVertex, duplicatedRightBankToVertex, verticesCount );
+  vertices = std::vector<Vertex>( verticesCount );
+
+  for ( const MDAL::DriverFlo2D::CellCenter &cell : cells )
+  {
+    std::map<size_t, size_t>::const_iterator it = cellsIdToVertex.find( cell.id );
+    if ( it != cellsIdToVertex.end() )
+    {
+      if ( it->second < vertices.size() )
+      {
+        Vertex &vertex = vertices[it->second];
+        //if coordinates of vertex is not nan, this coordinates is the one of the first bank, so add the second and use factor 0.5
+        if ( std::isnan( vertex.x ) )
+        {
+          vertex.x = cell.x;
+          vertex.y = cell.y;
+        }
+        else
+        {
+          vertex.x += cell.x;
+          vertex.x *= 0.5;
+          vertex.y += cell.y;
+          vertex.y *= 0.5;
+        }
+      }
+    }
+
+    std::map<size_t, std::vector<size_t>>::const_iterator itDuplicated = duplicatedRightBankToVertex.find( cell.id );
+    if ( itDuplicated != duplicatedRightBankToVertex.end() )
+    {
+      const std::vector<size_t> &indexes = itDuplicated->second;
+      for ( size_t index : indexes )
+      {
+        if ( index < vertices.size() )
+        {
+          Vertex &vertex = vertices[index];
+          if ( std::isnan( vertex.x ) )
+          {
+            vertex.x = cell.x;
+            vertex.y = cell.y;
+          }
+          else
+          {
+            vertex.x += cell.x;
+            vertex.x *= 0.5;
+            vertex.y += cell.y;
+            vertex.y *= 0.5;
+          }
+        }
+      }
```

Comment:
I do not understand the logic in this section. Can you please point me to the page in documentation explaining it?
What I see here is:
1. we parse the vertex locations from parseCHANBANKFile
2. in this code we do an average (*0.5) from vertex location and left cell location
3. and we do some average (*0.5) from the point 2 location and ALL duplicated right cell locations

If I understand it correctly, we want to get the "middle" line between tho banks (left and right). I do not undertand, 
1. first if we already have vertex coords, why we need to add left and right bank and then average it back? 
2. what is the relation between right, left bank, vertex, and cell?
3. if we are averaging for example 3 duplicates, the average is (p1 + p2 +p3) /3, but we do ((p1 + p2) / 2 + p3) / 2

Reply:
I have added some comments to explain the logic. Hope it is clearer.